Document presets for performance tuning to ES output

kilfoyle opened the pull request:
This adds two new sections to the Fleet [Elasticsearch output settings](https://ingest-docs_749.docs-preview.app.elstc.co/guide/en/fleet/master/es-output-settings.html) docs page:

1. Description of the new "Performance tuning" option:

    ![screen](https://github.com/elastic/ingest-docs/assets/41695641/fbe7c07e-9b40-4e39-a36d-855ee72c11c1)

2. These two [Performance tuning settings](https://ingest-docs_749.docs-preview.app.elstc.co/guide/en/fleet/master/es-output-settings.html#es-output-settings-performance-tuning-settings) tables, one showing the settings and one showing the potential performance impact.

Closes: #720

kpollich commented on docs/en/ingest-management/fleet/fleet-settings-output-elasticsearch.asciidoc:
```diff
@@ -81,6 +81,16 @@ output is set in the <<agent-policy,agent policy>>.
+[id="es-agent-performance-tuning"]
+**Performance tuning**
+
+| Choose one of the menu options to tune your {agent} performace when sending data to an {es} output. You can optimize for throughput, scale, latency, or you can choose a balanced (the default) set of performance specifications. Refer to <<es-output-settings-performance-tuning-settings,Performance tuning settings>> for details about the setting values and their potential impact on performance.
```
Typo fix

```suggestion
| Choose one of the menu options to tune your {agent} performance when sending data to an {es} output. You can optimize for throughput, scale, latency, or you can choose a balanced (the default) set of performance specifications. Refer to <<es-output-settings-performance-tuning-settings,Performance tuning settings>> for details about the setting values and their potential impact on performance.
```

kilfoyle replied:
Thanks for catching that! Fixed via [1df7941](https://github.com/elastic/ingest-docs/commit/1df7941f6e9ead235001dd4eb85d90e33389a2fe)

kpollich commented on docs/en/ingest-management/fleet/fleet-settings-output-elasticsearch.asciidoc:
```diff
@@ -81,6 +81,16 @@ output is set in the <<agent-policy,agent policy>>.
+
+| Choose one of the menu options to tune your {agent} performace when sending data to an {es} output. You can optimize for throughput, scale, latency, or you can choose a balanced (the default) set of performance specifications. Refer to <<es-output-settings-performance-tuning-settings,Performance tuning settings>> for details about the setting values and their potential impact on performance.
+
+You can also use the <<es-output-settings-yaml-config,Advanced YAML configuration>> field to set custom values. Note that if you adjust performance settings through the advanced YAML configuration, the **Performance tuning** option automatically changes to `Custom`.
```
I think it might be good to be more explicit that "performance settings" means "any of the settings that appear below". Maybe

> Note that if you adjust any of the performance settings below through the advanced YAML configuration, the Performance tuning option automatically changes to `Custom`, and cannot be changed.

kpollich reviewed (approved):
Overall LGTM just two minor comments.

kpollich commented:
@kilfoyle - Should we avoid documenting the much lower throughput numbers for serverless related to stateful? Maybe you know a good person to ask if we have an approach to things like this elsewhere in serverless docs. 

cc @joshdover @jlind23 maybe you have input as well.

jlind23 commented:
> cc @joshdover @jlind23 maybe you have input as well.

I expect serverless performance to drastically improve over the next weeks/months, so I would indeed avoid documenting them for now.

kilfoyle commented:
Thanks for the feedback @kpollich.

I've addressed the comments. For the "Potential performance impact" table, I removed the "Serverless throughput (relative to stateful)" row (via [1df7941](https://github.com/elastic/ingest-docs/pull/749/commits/1df7941f6e9ead235001dd4eb85d90e33389a2fe)). Here's how it looks now:

![Screenshot 2023-12-11 at 9 57 04 AM](https://github.com/elastic/ingest-docs/assets/41695641/89d9061c-16b5-47d8-a1a9-93031413a61f)

By the way, the security updates in this repo cause new commits to automatically dismiss any previous approvals. Would you mind approving again?

kpollich reviewed (approved):
Thanks for addressing changes! 🚀

kilfoyle commented:
@elasticmachine run elasticsearch-ci/docs

karenzone reviewed (approved):
LGTM pending passing docs-ci

kilfoyle commented:
@elasticmachine, run elasticsearch-ci/docs

kilfoyle commented:
@elasticmachine, run elasticsearch-ci/docs

lcawl commented:
@elasticmachine run elasticsearch-ci/docs

kilfoyle commented:
@elasticmachine run buildkite/docs-build-pr

jlind23 commented:
@elasticmachine run buildkite/docs-build-pr

kilfoyle commented:
I tried rebuilding directly from the [buildkite page](https://buildkite.com/elastic/docs-build-pr/builds/2617) but it's still failing. We're investigating...

kilfoyle commented:
run docs-build rebuild
